### PR TITLE
chore: setup Github actions

### PR DESCRIPTION
**Pull request by mapreal19**

### CODE REVIEW

File: .github/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [6.x, 8.x, 10.x]
```
Comment (graingert) [inline .github/main.yml:12]: no 9.0?

**Comment by graingert** [comment]
this won't run until there's a workflow on the default branch one mo

File: .github/main.yml (hunk 1 above)
Comment (mapreal19) [reply .github/main.yml]: I doubt anyone is using it? We should support 12 & 14

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,36 +1,21 @@
...
-        run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm install
```
Comment (graingert) [inline .github/workflows/main.yml:20]:
```suggestion
    - run: npm ci
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,36 +1,21 @@
...
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
+    strategy:
+      matrix:
+        node-version: [6.x, 8.x, 10.x]
```
Comment (graingert) [inline .github/workflows/main.yml:12]:
```suggestion
        node-version: [8.x, 10.x]
```

**Pull request merged**